test/jdk/java/lang/Math/Tests.java

```diff
@@ -37,6 +38,38 @@
 public class Tests {
     private Tests(){}; // do not instantiate
 
+    static volatile double zero = 0.0;
```

Comment:
Not directly related to this PR and just out of curiosity: is there a reason for this to be `volatile` and non-`final`?

Reply:
The comment on the PLATFORM_NAN field was mean to address that -- an attempt to make sure the expression to create the platform NaN gets evaluated at runtime rather than being constant-folded at compile time.